src/vending-machine.js

```diff
@@ -0,0 +1,61 @@
+const processData = require('./../__mock__/process-data');
+
+class vendingMachine {
+    constructor(data) {
+        this.data = processData(data);
+        // console.log(this.data)
+    }
+
+    printInvintory () {
+        return this.data.invintory;
+    }
+
+    refillInvintory () {
+        const refill = 30;
+        for( var i = 0; i < this.data.invintory.length; i++ ) {
+            if(this.data.invintory[i].amount <= 5) {
+                var resupplyItems = refill - this.data.invintory[i].amount;
+                return this.data.invintory[i].amount + resupplyItems;
+            }
+        }
+    }
+
+    resupplyChange () {
+        const resupplied = 100;
+        for( var i = 0; i < this.data.coins.length; i++ ) {
+            if(this.data.coins[i].amount <= 25) {
+                var resupplyAmount = resupplied - this.data.coins[i].amount;
+                // console.log(resupplyAmount);
+                return this.data.coins[i].amount + resupplyAmount;
+            }
+        }
+    }
+
+    itemDispense (balance) {
```

Comment:
In the future, think about how this actually works. You enter a balance, then you choose what you want dispensed. Always try to model your code after the real use case.